pkg/api/client/httpdoer.go

```diff
@@ -0,0 +1,97 @@
+package client
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// Doer performs HTTP requests.
+//
+// The standard http.Client implements this interface.
+type httpRequestDoer interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// Option allows setting custom parameters during construction
+type Option func(*componentInventoryClient) error
+
+// WithHTTPClient allows overriding the default Doer, which is
+// automatically created using http.Client. This is useful for tests.
+func WithHTTPClient(doer httpRequestDoer) Option {
+	return func(c *componentInventoryClient) error {
+		c.client = doer
+		return nil
+	}
+}
+
+// WithAuthToken sets the client auth token.
+func WithAuthToken(authToken string) Option {
+	return func(c *componentInventoryClient) error {
+		c.authToken = authToken
+		return nil
+	}
+}
+
+func (c *componentInventoryClient) get(ctx context.Context, path string) ([]byte, error) {
+	requestURL, err := url.Parse(fmt.Sprintf("%s%s", c.serverAddress, path))
+	if err != nil {
+		return nil, Error{Cause: err.Error()}
```

Comment:
This could have been `return nil, errors.Wrap("parsing URL", err.Error())` and I think we could avoid copying the FleetDB custom client error stuff.